File: p009.hs
{- 
 - Solution to Project Euler problem 9
 - By Nayuki Minase
 - 
 - http://nayuki.eigenstate.org/page/project-euler-solutions
 - https://github.com/nayuki/Project-Euler-solutions
 -}


p = 1000
main = putStrLn (show ans)
ans = head [a * b * (p - a - b) | a <- [1..p], b <- [1..p], isRight a b]
isRight a b = a <= b && b <= c
	&& a * a + b * b == c * c
	where c = p - a - b

File: p009.java
/* 
 * Solution to Project Euler problem 9
 * By Nayuki Minase
 * 
 * http://nayuki.eigenstate.org/page/project-euler-solutions
 * https://github.com/nayuki/Project-Euler-solutions
 */


public final class p009 implements EulerSolution {
	
	public static void main(String[] args) {
		System.out.println(new p009().run());
	}
	
	
	public String run() {
		for (int a = 1; a < 1000; a++) {
			for (int b = a + 1; b < 1000; b++) {
				int c = 1000 - a - b;
				if (a * a + b * b == c * c)  // Note: This implies b < c
					return Integer.toString(a * b * c);
			}
		}
		throw new AssertionError("Not found");
	}
	
}
